backend/api/events/events.py

```diff
@@ -36,6 +36,49 @@
 }
 
 
+@api.get("/paginate", tags=["Events"])
+def list_events(
+    subject: User = Depends(registered_user),
+    event_service: EventService = Depends(),
+    order_by: str = "time",
+    ascending: str = "true",
+    filter: str = "",
+    range_start: str = "",
+    range_end: str = "",
+) -> Paginated[EventDetails]:
+    """List events in time range via standard backend pagination query parameters."""
+
+    pagination_params = EventPaginationParams(
+        order_by=order_by,
+        ascending=ascending,
+        filter=filter,
+        range_start=range_start,
+        range_end=range_end,
+    )
+    return event_service.get_paginated_events(pagination_params, subject)
+
+
+@api.get("/paginate/unauthenticated", tags=["Events"])
+def list_events_unauthenticated(
+    event_service: EventService = Depends(),
+    order_by: str = "time",
+    ascending: str = "true",
+    filter: str = "",
+    range_start: str = "",
+    range_end: str = "",
+) -> Paginated[EventDetails]:
+    """List events in time range via standard backend pagination query parameters for unauthenticated users."""
+
+    pagination_params = EventPaginationParams(
+        order_by=order_by,
+        ascending=ascending,
+        filter=filter,
+        range_start=range_start,
+        range_end=range_end,
+    )
+    return event_service.get_paginated_events(pagination_params)

```

Comment:
Just curious (because I am not familiar with the pagination code as much), do we need to have a separate `/paginate/unauthenticated` API route, or can we just pass in `subject` into the `/paginate` API and check if the user is unauthenticated that way?

Reply:
Yep it does need a separate route. I'm not sure how it works, but the API route auth vs. not auth takes into account any path that needs authentication I think. So, if there is a subject parameter, it will require authentication.